docs/index.rst

```diff
@@ -101,19 +101,22 @@ Basic usage
 
 .. testcode::
 
-   class MLP(nn.Module):
+   class MLP(nn.Module):                    # dataclass Modules
+     out_dims: int                         
+
      @nn.compact
      def __call__(self, x):
-       x = nn.Dense(16)(x)                # inline submodules
-       x = nn.relu(x)
-       x = nn.Dense(16)(x)                # inline submodules
+       x = x.reshape((x.shape[0], -1))       
+       x = nn.Dense(128)(x)                 # create submodules inline
+       x = nn.relu(x)                       
+       x = nn.Dense(self.out_dims)(x)       # shape inference
        return x
 
-   model = MLP()                          # create model
+   model = MLP(out_dims=10)                 # create model
 
-   x = jnp.ones((4, 16))                  # get some data
-   variables = model.init(PRNGKey(42), x) # initialize weights
-   y = model.apply(variables, x)          # make forward pass
+   x = jnp.empty((4, 28, 28, 1))            # fake data
+   variables = model.init(PRNGKey(42), x)   # initialize weights
```

Comment:
Suggestion: Add "the" (ref: https://www.deeplearning.ai/ai-notes/initialization/index.html)

```suggestion
   variables = model.init(PRNGKey(42), x)   # initialize the weights
```